// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message IapIdentityAwareProxyClient {
  string name = 1;
  string secret = 2;
  string display_name = 3;
  string project = 4;
  string brand = 5;
}

message ApplyIapIdentityAwareProxyClientRequest {
  IapIdentityAwareProxyClient resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteIapIdentityAwareProxyClientRequest {
  string service_account_file = 1;
  IapIdentityAwareProxyClient resource = 2;
}

message ListIapIdentityAwareProxyClientRequest {
  string service_account_file = 1;
  string Project = 2;
  string Brand = 3;
}

message ListIapIdentityAwareProxyClientResponse {
  repeated IapIdentityAwareProxyClient items = 1;
}

service IapIdentityAwareProxyClientService {
  rpc ApplyIapIdentityAwareProxyClient(ApplyIapIdentityAwareProxyClientRequest) returns (IapIdentityAwareProxyClient);
  rpc DeleteIapIdentityAwareProxyClient(DeleteIapIdentityAwareProxyClientRequest) returns (google.protobuf.Empty);
  rpc ListIapIdentityAwareProxyClient(ListIapIdentityAwareProxyClientRequest) returns (ListIapIdentityAwareProxyClientResponse);
}
